doc/source/ray-air/examples/automl_with_ray_air.ipynb

```diff
@@ -0,0 +1,858 @@
+    "    assert Y_train_df is not None, \"Must pass in the dataset\"\n",
+    "\n",
+    "    # Get the model class\n",
+    "    model_cls, model_params = config.get(\"model_cls_and_params\")\n",
+    "    freq = config.get(\"freq\")\n",
+    "    metrics = config.get(\"metrics\", {\"mse\": mean_squared_error})\n",
+    "\n",
+    "    # CV params\n",
+    "    test_size = config.get(\"test_size\", None)\n",
+    "    n_splits = config.get(\"n_splits\", 5)\n",
+    "\n",
+    "    model = model_cls(**model_params)\n",
+    "\n",
+    "    # Default the parallelism to the # of cross-validation splits\n",
+    "    parallelism_kwargs = {\"n_jobs\": n_splits}\n",
+    "\n",
+    "    # Initialize statsforecast with the model\n",
+    "    statsforecast = StatsForecast(\n",
+    "        df=Y_train_df,\n",
+    "        models=[model],\n",
+    "        freq=freq,\n",
+    "        **parallelism_kwargs,\n",
+    "    )\n",
+    "\n",
+    "    # Perform temporal cross-validation\n",
+    "    test_size = test_size or len(Y_train_df) // (n_splits + 1)\n",
+    "\n",
+    "    # Example: using the default test_size\n",
+    "    # Dataset with 12 items = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]\n",
+    "    # n_splits = 5\n",
+    "    # test_size = 12 // (5 + 1) = 2\n",
+    "    # Splits:\n",
+    "    # Train = [0, 1]   Test = [2, 3]\n",
+    "    # Train = [0, 1, 2, 3]   Test = [4, 5]\n",
+    "    # Train = [0, 1, 2, 3, 4, 5]   Test = [6, 7]\n",
+    "    # Train = [0, 1, 2, 3, 4, 5, 6, 7]   Test = [8, 9]\n",
+    "    # Train = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]   Test = [10, 11]\n",
+    "\n",
+    "    start_time = time.time()\n",
+    "    forecasts_cv = statsforecast.cross_validation(\n",
+    "        h=test_size,\n",
+    "        n_windows=n_splits,\n",
+    "        step_size=test_size,\n",
+    "    )\n",
+    "    cv_time = time.time() - start_time\n",
+    "\n",
+    "    # Compute metrics (according to `metrics`)\n",
+    "    cv_results = compute_metrics_and_aggregate(forecasts_cv, model, metrics)\n",
+    "\n",
+    "    # Report metrics and save cross-validation output DataFrame\n",
+    "    results = {\n",
+    "        **cv_results,\n",
+    "        \"cv_time\": cv_time,\n",
+    "    }\n",
+    "    checkpoint_dict = {\n",
+    "        \"cross_validation_df\": forecasts_cv,\n",
+    "    }\n",
+    "    checkpoint = Checkpoint.from_dict(checkpoint_dict)\n",
+    "    session.report(results, checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5cf6e512",
+   "metadata": {},
+   "source": [
+    "We also define a helper function used above, which calculates the metrics from the dataframe output of `StatsForecast.cross_validation`. For example, we will calculate the mean squared error between the model predictions and the actual observed data, averaged over all training splits. This metric gets reported to Tune as `mse_mean`, which is the metric we will use to define the best-performing model."
```

Comment:
The flow here is a bit weird; you probably to
- move the docstring from the function below, out of the cell
- move this text above the previous cell
- use the text box above the previous cell to describe both functions, something like:

```
we will define two methods below:

- cross_validation, which does X
- compute_metrics_and_aggregate which is a helper function used in cross_validation and does ABC